python/cugraph/cugraph/experimental/community/triangle_count.py

```diff
@@ -89,11 +89,11 @@ def EXPERIMENTAL__triangle_count(G, start_list=None):
 
     srcs = G.edgelist.edgelist_df['src']
     dsts = G.edgelist.edgelist_df['dst']
-    weights = G.edgelist.edgelist_df['weights']
 
-    if srcs.dtype != 'int32':
-        raise ValueError(f"Graph vertices must have int32 values, "
-                         f"got: {srcs.dtype}")
+    if (G.is_weighted() is False):
+        weights = cudf.Series([1.0]).repeat(G.edgelist.edgelist_df['src'].size)
```

Comment:
I think you can undo the changes to this file. I am working on a PR that will move `triangle_count` from `experimental` to `promoted` and I will also enable it to leverage the `PLC graph` which will internally create the weight column if it doesn't exist as it can be seeing [here](https://github.com/rapidsai/cugraph/blob/branch-22.10/python/cugraph/cugraph/structure/graph_implementation/simpleGraph.py#L771)